pkg/jsonutil/conversion.go

```diff
@@ -67,6 +68,15 @@ func ProtoToJSON(m protocompat.Message, options ...ConversionOption) (string, er
 		return "", err
 	}
 
+	if contains(options, OptCompact) {
+		compactBuf := &bytes.Buffer{}
+		if err := json.Compact(compactBuf, x); err != nil {
+			return string(x), nil
```

Comment:
Maybe add a comment here stating that in case the JSON compaction fails, the JSON buffer is returned (best effort).